pkg/controller/proxyconfig/validation.go

```diff
@@ -0,0 +1,53 @@
+package proxyconfig
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-network-operator/pkg/util/validation"
+)
+
+const (
+	proxyHTTPScheme = "http"
+	// noProxyWildcard is the string used to as a wildcard attached to a
+	// domain suffix in proxy.spec.noProxy to bypass proxy for httpProxy
+	// and httpsProxy.
+	noProxyWildcard = "*"
+)
+
+// ValidateProxyConfig ensures that proxyConfig is valid.
+func (r *ReconcileProxyConfig) ValidateProxyConfig(proxyConfig *configv1.ProxySpec) error {
```

Comment:
(possible follow-up todo): Do we want to check if the specified proxy is reachable?

Reply:
@squeed prior to my last commit (`efca2437f3b756f2d92613bc15221db2755c5306`), this PR included support for `readinesCheck` which accomplished your ask. However, it was decided to have this PR simply validate the core `http/https/no proxy` proxy fields and write to status. This PR will provide a functional proxy environment so other groups are not blocked by this work. I'll provide a follow-on PR shortly that includes `readinesCheck` support.

Reply:
:+1: